# Chapter 1: Numbers, Equations, and Inequalities
***

### Table of Contents

1. [Exercise on the multiplication table](#1.-Exercise-on-the-multiplication-table)
2. [An interesting property of numbers (WIP)](#2.-An-interesting-property-of-numbers-(WIP))
3. [Division by 11](#3.-Division-by-11)
4. [The divisibility of numbers](#4.-The-divisibility-of-numbers)
5. [A simplified form of Fermat's theorem](#5.-A-simplified-form-of-Fermat's-theorem)

## 1. Exercise on the multiplication table
***
Construct sequence of numbers as follows:

$2 \cdot 3 = 6$

$3 \cdot 6 = 18$

$6 \cdot 1 = 6$, $1 \cdot 8 = 8$

Sequence: $2,3,6,1,8,6,8...$

**Prove** that numbers $5, 7, 9$ never appear in this sequence

#### Attempt:

We list the conditions for 5, 7, 9 to appear and check each one. All the ways for 5 to appear:

To get _5
- 1*5 = 5
- 3*5 = 15
- 5*5 = 25
- 7*5 = 35
- 9*5 = 45

Immediately, we have a problem. 5 is a prime number, so we need a 5 to generate _5, which is not possible in this sequence that starts with 2 and 3.

To get 5_, find factorial s.a. all are single digit
- 51 = 3*17
- 52 = 2 * 3 * 13
- 53 = 1 * 53
- **54 = 3 * 2 * 9 -> 6 * 9**
- 55 = 5 * 11
- **56 = 2 * 2 * 2 * 7 -> 8 * 7**
- 57 = 3 * 19
- 58 = 2 * 29
- 59 = 1 * 59

This requires the presence of 9 or 7. We repeat the exercise:

To get_7
- 1*7 = 7
- 3*9 = 27

To get 7_
- 72 = 8*9

Again since 7 is a prime number, we need a 9 to yield 27 or 72.

To get _9
- 1*9 = 9
- 3*3 = 9
- 7*7 = 49

To get 9_ is impossible given highest multiplication table of single digits is 81.

For 1 * 9, it is impossible because we are generating the first instance of 9.

For 3 * 3, we need either:
- 33, with factors 3 and 11, but 11 is a double digit
- _ 3.3 _ 

_ 3 must be 3 and not 63 because 63 requires 7 and 9. 3_ is either 3 or 32 (4 * 8). 3 is impossible because it is a prime that requires 3 * 1 and the only number after single 3 is 6 (2*3) or 2 (for 32). It is therefore impossible to follow a single 3 with 32. 

For 7 * 7, we need the number 7 which requires 9 so that's also not possible.

Since there cannot be 9, and 5 and 7 depend on 9, there also cannot be 5 or 7. So 5, 7, 9 never appears in this sequence.

#### Book Solution:

Starts with proving that the sequence cannot have two odd numbers in a row.

Let a b c d be an instance of the sequence for this exercise. 

Suppose a * b = cd. If cd is odd, then a and b must both be odd. c can be either odd or even.

Supose a * b = c. If c is odd, then a and b must both be odd.

In either case, the immediately two preceding terms must be odd. The first three sequences must either be:
- odd odd odd 
- odd odd even

The three starting digits in this sequence is 2, 3, and 6 which is even, odd, even. Therefore there cannot be any consecutive odd numbers.

It follows that the number 9 cannot appear as a product of 3 * 3. Also, numbers larger or equal to 90 are impermissible because the product requires one number with two digits e.g. 9 * 10. 

The number 7 requires the number 9 (8 * 9 = 72), and the number 5 requires either 7 or 9 (6 * 9 = 54 or 7 * 8 = 56). Since number 9 cannot appear, it follows that number 7 cannot appear, and number 5 cannot appear.


#### Code

In [1]:
class Q1():
    @staticmethod
    def split_number(n: int) -> list[int]:
        return [int(x) for x in str(n)] 

    # count is number of multiplications
    @classmethod
    def create_sequence(cls, a: int, b: int, count: int) -> list[int]:
        lst = [a, b]
        i = 0
        while i < count: 
            c = lst[i] * lst[i+1]
            lst.extend(cls.split_number(c))
            i += 1
        return lst

    @staticmethod
    def is_in_any(n_lst: int, seq_lst: list) -> bool:
        return any([x for x in list(set(seq_lst)) if x in n_lst])

    @staticmethod
    def is_in_all(n_lst: int, seq_lst: list) -> bool:
        return set(n_lst) < set(seq_lst)

    @staticmethod
    def count_nums(seq_lst: list) -> int:
        nums = [n for n in range(1,10)]
        total = len(seq_lst)
        for n in nums:
            count = len([x for x in seq_lst if x == n])
            print(f'for {n} in seq start {seq_lst[:2]}: {"%.0f" % (100*count/total)}%, {count}/{total}')

    @classmethod
    def print_solution(cls, seq):
        sub_seq1 = [5,7,9]
        sub_seq2 = [2,4,6]
        print(f'''
        for sequence {seq[0:10]}... of length {len(seq1)}
        any {sub_seq1} is in sequence: {cls.is_in_any(sub_seq1, seq)}
        any {sub_seq2} is in sequence: {cls.is_in_any(sub_seq2, seq)}
        all {sub_seq1} is in sequence: {cls.is_in_all(sub_seq1, seq)}
        all {sub_seq2} is in sequence: {cls.is_in_all(sub_seq2, seq)}
        check unique numbers of sequence: {set(seq)}
        ''')
        cls.count_nums(seq)

In [2]:
s1 = Q1()

In [3]:
seq1 = s1.create_sequence(a=2, b=3, count=10000)
seq2 = s1.create_sequence(a=3, b=7, count=10000)

In [4]:
s1.print_solution(seq1)
s1.print_solution(seq2)


        for sequence [2, 3, 6, 1, 8, 6, 8, 4, 8, 4]... of length 16747
        any [5, 7, 9] is in sequence: False
        any [2, 4, 6] is in sequence: True
        all [5, 7, 9] is in sequence: False
        all [2, 4, 6] is in sequence: True
        check unique numbers of sequence: {1, 2, 3, 4, 6, 8}
        
for 1 in seq start [2, 3]: 4%, 674/16747
for 2 in seq start [2, 3]: 27%, 4443/16747
for 3 in seq start [2, 3]: 2%, 317/16747
for 4 in seq start [2, 3]: 35%, 5931/16747
for 5 in seq start [2, 3]: 0%, 0/16747
for 6 in seq start [2, 3]: 17%, 2773/16747
for 7 in seq start [2, 3]: 0%, 0/16747
for 8 in seq start [2, 3]: 16%, 2609/16747
for 9 in seq start [2, 3]: 0%, 0/16747

        for sequence [3, 7, 2, 1, 1, 4, 2, 1, 4, 8]... of length 16747
        any [5, 7, 9] is in sequence: True
        any [2, 4, 6] is in sequence: True
        all [5, 7, 9] is in sequence: False
        all [2, 4, 6] is in sequence: True
        check unique numbers of sequence: {1, 2, 3, 4, 6, 7, 8}
    

## 2. An interesting property of numbers (WIP)
***
Add the squares of the digits of an arbitrary number e.g. 2583

$2^2+5^2+8^2+3^2 = 102$

Repeat with the new number obtained

- $1^2+0^2+2^2=5$
- $5^2=25$
- $2^2+5^2=29$
- $2^2+9^2=85$
- $8^2+5^2=89$
- $8^2+9^2=145$

And so on.

**Prove** that this procedure either leads to the number 1 (and therefore recurr indefinitely at $1^2=1$), or to the number 145 with the following sequence repeated indefinitely

$$145, 42, 20, 4, 16, 37, 58, 89$$

#### Attempt

Again, exploring single digits, but now with squares. Let's see if we can tell what's so special from listing the squares.

$0^2 = 0$   
$1^2 = 1$   
$2^2 = 4$   
$3^2 = 9$   
$4^2 = 16$  
$5^2 = 25$  
$6^2 = 36$  
$7^2 = 49$  
$8^2 = 64$  
$9^2 = 81$  

The only way to end up with $1$ is with $1 \cdot 10^n$, so $1^2 + 0^2 + 0^2 + ... + 0^2 = 1$

E.g. $5555$ gives 
- $5^2 + 5^2 + 5^2 + 5^2 = 100$
- $1^2 + 0^2 + 0^2 = 1$

We get $145 = 8^2 + 9^2 = 2^{3\cdot2} + 3^{2\cdot2} = 10^2\cdot1 + 10^1\cdot4 + 10^0\cdot5$

Highest is $9^2+9^2=162$
- $1^2+6^2+2^2=41$
- $4^2+1^2=17$
- $1^2+7^2=50$
- $5^2+0^2=25$
- $2^2+5^2=29$
- $2^2+9^2=85$
- $8^2+5^2=89$
- $8^2+9^2=145$

... I have no clue

### Book Solution

**First, establish that an arbitrary number with >=3 digits is greater in value than the sum of its squared digits.**

Let arbitrary number be $L = 10^{n-1}a_n + 10^{n-2}a_{n-1} + ... + 10^2a_3 + 10a_2 + a_1$ where $n=1,2,3...$ and $a_n=0,1,2,...,9$

Let sum of its squared digits of L be $L_s = a^2_n + a^2_{n-1} + ... + a^2_3 + a^2_2+ a^2_1$

$L-L_s = (10^{n-1}-a_n)a_n + (10^{n-2}-a_{n-1})a_{n-1} + ... + (10^2-a_3)a_3 + (10^1-a_2)a_2 - (a_1-1)a_1$

It follows that $(a_1-1)a_1 \le 72$, since max $a_1 = 9$

For $n\ge3$ and $a_n\neq0$, we have $(10^{n-1}-a_n)a_n\ge99$, for example for $n=3, a_n=1$, we have $(100-1)1\ge99$

For $n\ge2$ and $a_n\neq0$, we have $(10^{n-1}-a_n)a_n\ge0$, for example for $n=2, a_n=1$, we have $(10-1)1\ge0$

It follows that $L-L_s\ge99-72$ or $L-L_s\ge27$. Therefore we have the sequence $L_1 > L_2 > L_3 ...$ We have proved that for numbers **greater than 3 digits**, the original number will always be greater in value to the sum of its squared digits.

**Second, establish the same for numbers <3 digits**

Rewriting the inequality for n=3, $L-L_s\ge27$ means there is at least one number in the sequence of $L_1, L_2, L_3...$ that is a double digit number, given $L > L_1 > L_2 ... $. Recall that starting with triple digit number, our sequence will get smaller and smaller. 

Let that number be $L_q = 10j+k$ with sequence $L_{q+1}, L_{q+2}, L_{q+3}...$

Refer to table outlined in the book to validate each possible number of $j^2+k^2$

#### Code

In [5]:
class Q2():
    @staticmethod
    def create_seq(n: int) -> list[int]:
        lst_split=Q1.split_number(n)
        n_new=sum([x**2 for x in lst_split])
        print(n_new)
        if n_new==145 or n_new==1:
            return None
        return Q2.create_seq(n_new)

In [6]:
import random

In [7]:
seed_num = random.randint(0,1e6)
print(f'seed: {seed_num}')
Q2.create_seq(seed_num)

seed: 174086
166
73
58
89
145


In [8]:
Q2.create_seq(4)

16
37
58
89
145


## 3. Division by 11
***

Prove that the number $5^{5k+1} + 4^{5k+2} + 3^{5k}$ is divisible by 11 for every natural k.

#### Attempt

#### Book Solution

Look at remainders!! Of course.

Rewrite above expression as $5^{\alpha} + 4^{\beta} + 3^{\gamma}$. Dividing each term by 11, we get corresponding remainders $R(5^{\alpha}), R(4^{\beta}), R(3^{\gamma})$.

We want $R(5^{\alpha}) + R(4^{\beta}) + R(3^{\gamma})$ to be a multiple of 11. From table below, $\alpha=6, \beta=7, \gamma=0$ will yield a sum of $5+5+1=11$.

From the table, we see that's $5^6 + 4^7 + 3^0$, or equivalently $5^{5k+1} + 4^{5k+2} + 3^{5k}$ when $k=1$. 

For all other $k=2,3,4...$ results in same pattern of remainders, therefore $5^{5k+1} + 4^{5k+2} + 3^{5k}$ is divisible by 11 for all natural $k$.

There can be other solutions as well, e.g. $5^5+4^3+3^5$ is also divisible by 11 because it gives sum of remainders $1+9+1=11$.


In [9]:
for n in [5,4,3]:
    [print(f'{n}^{x} %11 {(n**x)%11}') for x in range(0,10)]

5^0 %11 1
5^1 %11 5
5^2 %11 3
5^3 %11 4
5^4 %11 9
5^5 %11 1
5^6 %11 5
5^7 %11 3
5^8 %11 4
5^9 %11 9
4^0 %11 1
4^1 %11 4
4^2 %11 5
4^3 %11 9
4^4 %11 3
4^5 %11 1
4^6 %11 4
4^7 %11 5
4^8 %11 9
4^9 %11 3
3^0 %11 1
3^1 %11 3
3^2 %11 9
3^3 %11 5
3^4 %11 4
3^5 %11 1
3^6 %11 3
3^7 %11 9
3^8 %11 5
3^9 %11 4


In [10]:
for k in [k for k in range(1,10)]:
    num = 5**(5*k+1) + 4**(5*k+2) + 3**(5*k)
    divisor=11
    result = num%divisor==0
    print(f'k={k}, {num} is divisible by {divisor}: {result}')

k=1, 32252 is divisible by 11: True
k=2, 65664390 is divisible by 11: True
k=3, 169782108716 is divisible by 11: True
k=4, 494432831031942 is divisible by 11: True
k=5, 1508131365182857052 is divisible by 11: True
k=6, 4675059823042234224390 is divisible by 11: True
k=7, 14570804744329875486495116 is divisible by 11: True
k=8, 45494077913917911421604180742 is divisible by 11: True
k=9, 142128354195602915965174073201852 is divisible by 11: True


We can repeat this for other divisions, e.g. solve for $2^{\alpha} + 3^{\beta} + 4^{\gamma}$ such that the sum is divisible by 3. We print table (below) and solve, e.g. $2^{3} + 3^{0} + 4^{1}$ is divisible by 3 because the sum of remainder of each term divided by 3 is a multiple of 3. 

The expression $2^{2k} + 3^{k-k} + 4^{k}$ is divisible by 3 for all natural $k$.

In [11]:
for n in [2,3,4]:
    [print(f'{n}^{x} %3 {(n**x)%3}') for x in range(0,10)]

2^0 %3 1
2^1 %3 2
2^2 %3 1
2^3 %3 2
2^4 %3 1
2^5 %3 2
2^6 %3 1
2^7 %3 2
2^8 %3 1
2^9 %3 2
3^0 %3 1
3^1 %3 0
3^2 %3 0
3^3 %3 0
3^4 %3 0
3^5 %3 0
3^6 %3 0
3^7 %3 0
3^8 %3 0
3^9 %3 0
4^0 %3 1
4^1 %3 1
4^2 %3 1
4^3 %3 1
4^4 %3 1
4^5 %3 1
4^6 %3 1
4^7 %3 1
4^8 %3 1
4^9 %3 1


In [12]:
for k in [k for k in range(1,10)]:
    num = 2**(2*k) + 3**(k-k) + 4**(k)
    divisor=3
    result = num%divisor==0
    print(f'k={k}, {num} is divisible by {divisor}: {result}')

k=1, 9 is divisible by 3: True
k=2, 33 is divisible by 3: True
k=3, 129 is divisible by 3: True
k=4, 513 is divisible by 3: True
k=5, 2049 is divisible by 3: True
k=6, 8193 is divisible by 3: True
k=7, 32769 is divisible by 3: True
k=8, 131073 is divisible by 3: True
k=9, 524289 is divisible by 3: True


## 4. The divisibility of numbers
***

The number $3^{105} + 4^{105}$ is divisible by 13, 49, 181, and 379, and is not divisible either by 5 or 11. 

How can this result be confirmed?

#### Attempt

We can check the results with code, but that's probably cheating considering the book, so let's reason through it first.

The sum of the remainders of each term in the expression divided by the divisor must be a multiple of that divisor in order for the results obtained by the expression to be divisible by that divisor.

In other words, for $R_n(3^\alpha + 4^\beta) = 0$ to be true where $n$ is the divisor and $R_n()$ is the remainder from $n$, the following must be true: $R_n(3^\alpha) + R_n(4^\beta) = k\cdot n$ for all natural $k$

However $3^{105}$ and $4^{105}$ are really large numbers, and not feasible to compute remainder directly assuming we are just pencil and paper, is there an easier way?

In [13]:
#brute force
divisors = [5, 11, 13, 49, 181, 379]
num = 3**105+4**105
for d in divisors:
    result = num%d==0
    print(f'{"%0.2e" % num} is divisible by {d}: {result}')

1.65e+63 is divisible by 5: False
1.65e+63 is divisible by 11: False
1.65e+63 is divisible by 13: True
1.65e+63 is divisible by 49: True
1.65e+63 is divisible by 181: True
1.65e+63 is divisible by 379: True


### Book Solution

$a^n+b^n$ is divisible by $a+b$ if $n$ is odd. (? I need to prove this first.)

$$\frac{a^n+b^n}{a+b}$$  
$$= \frac{(a^n+b^n)(a-b)}{(a+b)(a-b)}$$   
$$= \frac{a^{n+1}-a^nb+b^na+b^{n+1}}{a^2-b^2}$$   
$$= \frac{a^{n+1}+b^{n+1}}{a^2-b^2} - \frac{ab(a^{n-1}-b^{n-1})}{a^2-b^2}$$   
$$= \frac{(a^{\frac{n+1}{2}}+b^{\frac{n+1}{2}})(a^{\frac{n+1}{2}}-b^{\frac{n+1}{2}})}{a^2-b^2} - \frac{ab(a^{\frac{n-1}{2}}-b^{\frac{n-1}{2}})}{a^2-b^2}$$   

Recall $$a^k - b^k = (a-b)\cdot(a^{k-1} + a^{k-2}b + a^{k-3}b^2 + ...+ b^{k-1})$$
$$a^k + b^k = (a+b)\cdot(a^{k-1} - a^{k-2}b + a^{k-3}b^2 - ...+ b^{k-1})$$

If $n$ is odd, then the result will be an expression without fractions, because $a^{\frac{n-1}{2}}-b^{\frac{n-1}{2}}$ can be expressed in the form of $(a^k+b^k)(a^k-b^k)$, cancelling the denominator.

On the other hand, if $n$ is even, then  $a^{\frac{n-1}{2}}-b^{\frac{n-1}{2}}$ cannot be simplified to cancel the denominator, resulting in a fraction, and hence a remainder.

Since we have $3^{105}+4^{105}$, and $105$ is odd, it will be divisible by 7:

In [14]:
#brute force
divisor=7
num = (3**105+4**105)
result = num%divisor==0
print(f'{"%0.2e" % num} is divisible by {d}: {result}')

1.65e+63 is divisible by 379: True


In [15]:
[print(f'{3**x+4**x}') for x in range(0,10) if x%2!=0]

7
91
1267
18571
281827


[None, None, None, None, None]

In [16]:
[7*x for x in [5, 11, 13, 49, 181, 379]]

[35, 77, 91, 343, 1267, 2653]

In [17]:
[49*x for x in [5, 11, 13, 49, 181, 379]]

[245, 539, 637, 2401, 8869, 18571]

Now onto **13, 49, 181, and 379.**

$3^{105}+4^{105} = 3^{3\cdot35}+4^{3\cdot35} = (3^3)^{35}+(4^3)^{35}$

$3^3+4^3=91=7\cdot13$

$3^{105}+4^{105} = 3^{5\cdot21}+4^{5\cdot21} = (3^5)^{21}+(4^5)^{21}$

$3^5+4^5=1267=7\cdot181$

$3^{105}+4^{105} = 3^{7\cdot15}+4^{7\cdot15} = (3^7)^{15}+(4^7)^{15}$

$3^7+4^7=18571=7\cdot7\cdot379$

For rest of the numbers **(5, 11)** we use modulo operations. Recall

$$A+B\mod C = (A \mod C  +  B \mod C) \mod C$$    

From table below we get 
$$(4^2)^{52} \equiv (1)^{52} \pmod{5}$$
$$4^{104} \equiv 1 \pmod{5}$$ 
$$4^{105} \equiv 4 \pmod{5}$$   

Similarly, $3^2 \equiv -1 \pmod{5}$

$$(3^2)^{52} \equiv (-1)^{52} \pmod{5}$$ 
$$3^{104} \equiv 1 \pmod{5}$$
$$3^{105} \equiv 3 \pmod{5}$$   

Add the expressions together

$$4^{105} + 3^{105} \mod{5} = (4+3) \mod{5} = 2$$ 
$$4^{105} + 3^{105} \equiv 2 \pmod{5} $$   


Dividing by 5 gives us a remainder of 2. 

For 11, we repeat the process
$$(4^5)^{21} \equiv (1)^{21} \pmod{11}$$
$$4^{105} \equiv 1 \pmod{11}$$  
$$(3^5)^{21} \equiv (1)^{21} \pmod{11}$$
$$3^{105} \equiv 1 \pmod{11}$$   

Add the expressions together

$$4^{105} + 3^{105} \mod{11} = (1+1) \mod{11} = 2$$ 
$$4^{105} + 3^{105} \equiv 2 \pmod{11} $$   


In [18]:
for n in [3,4]:
    [print(f'{n}^{x} = {n**x}, %5 {n**x%5}') for x in range(0,10)]

3^0 = 1, %5 1
3^1 = 3, %5 3
3^2 = 9, %5 4
3^3 = 27, %5 2
3^4 = 81, %5 1
3^5 = 243, %5 3
3^6 = 729, %5 4
3^7 = 2187, %5 2
3^8 = 6561, %5 1
3^9 = 19683, %5 3
4^0 = 1, %5 1
4^1 = 4, %5 4
4^2 = 16, %5 1
4^3 = 64, %5 4
4^4 = 256, %5 1
4^5 = 1024, %5 4
4^6 = 4096, %5 1
4^7 = 16384, %5 4
4^8 = 65536, %5 1
4^9 = 262144, %5 4


In [19]:
for n in [3,4]:
    [print(f'{n}^{x} = {n**x}, %11 {n**x%11}') for x in range(0,10)]

3^0 = 1, %11 1
3^1 = 3, %11 3
3^2 = 9, %11 9
3^3 = 27, %11 5
3^4 = 81, %11 4
3^5 = 243, %11 1
3^6 = 729, %11 3
3^7 = 2187, %11 9
3^8 = 6561, %11 5
3^9 = 19683, %11 4
4^0 = 1, %11 1
4^1 = 4, %11 4
4^2 = 16, %11 5
4^3 = 64, %11 9
4^4 = 256, %11 3
4^5 = 1024, %11 1
4^6 = 4096, %11 4
4^7 = 16384, %11 5
4^8 = 65536, %11 9
4^9 = 262144, %11 3


In [20]:
def gen_factors(n:int, l:list[int]) -> list[int]:
    if n == 1:
        return None
    i=2
    while n%i!=0:
        i+=1
    l.append(i)
    if i<=n//2:
        gen_factors(n/i,l)
        return l

In [21]:
def print_factors(num: int) -> None:
    f_lst=gen_factors(num,[])
    print(f'factoring {num}... {f_lst}')  
    try:
        print(f'unique factors {set(f_lst)}')   
    except: 
        TypeError
    

In [22]:
num=random.randint(0,50)
print_factors(num)

factoring 31... None


In [23]:
print_factors(105)

factoring 105... [3, 5, 7]
unique factors {3, 5, 7}


## 5. A simplified form of Fermat's theorem
***
If $x,y,z,n$ are natural numbers, and $n\ge z$, then the relation $x^n+y^n=z^n$ does not hold.

#### Attempt

We know some classic relations like $3^2+4^2=5^2$ holds. 

What's the relationship of $x,y,z$? Intuitively $x<z, y<z$. Then $x \neq y$ else $x^n=y^n=z^n$. What can we use this relationship for in the case of $n \ge z$?

#### Book Solution

Uses contradiction to prove the hypothesis.

$$z^n - y^n = (z-y)\cdot(z^{n-1}+yz^{n-2}+...+y^{n-1}) \ge 1\cdot nx^{n-1} > x^n$$ 

Where did $\ge 1\cdot nx^{n-1}$ come from?

The book doesn't say but I think it could be the following reason:

First, $(z-y) \ge 1$ since we already established $y < z$ and all $x,y,z,n$ are natural numbers.
 
Second, to show $(z^{n-1}+yz^{n-2}+...+y^{n-1}) \ge nx^{n-1}$ assume for now $y = z$. We can rewrite as follows:

$$(z^{n-1}+z\cdot z^{n-2}+...+z^{n-1}) = nz^{n-1}$$

Given $x < z$, we conclude $nz^{n-1} \ge nx^{n-1}$. Since we have $y < z$ and $x \neq y$, we can assume $x < y < z$ (or we can switch x and y, just for convenience.) Therefore 

$$(z^{n-1}+yz^{n-2}+...+y^{n-1}) \ge nx^{n-1}$$

Since $n \ge z$ and therefore $n > x$, it follows that $1\cdot nx^{n-1} > x^n$

The expression $z^n - y^n > x^n$ written as $z^n > x^n + y^n$ clearly contradicts the premise that  $z^n = x^n + y^n$.


## 6. Distribution of numbers (WIP)
***

Find ten numbers $x_1, x_2,...,x_{10}$ such that
1. the number $x_1$ is contained in the closed interval [$0,1$]
2. the numbers $x_1, x_2$ lie in different halves of the interval [$0,1$]
3. the numbers $x_1, x_2, x_3$ lie each in different thirds of the interval [$0,1$] 
4. repeat ... until
5. the numbers $x_1, x_2,...x_{10}$ lie each in different tenth of the interval [$0,1$]

#### Attempt

Split solution into:
1. order of variables
2. value (range) of variables

Earlier variables have the most conditions:
- x1: 10 conditions
- x2: 9 conditions
- ...
- x9: 2 conditions
- x10: 1 condition

Therefore we need to solve for placement of earlier variables first.

The biggest danger is having earlier variables grouped too closely together, so we prioritize increasing the distance between xn and its immediate neighbors to be greater than 1. e.g. for [1,3,2] must become [1,_,3,_,2] as soon as possible. With every insertion following this rule we can be certain that prior conditions are fulfilled. This method yields for example

Sequence A: [1, 9, 5, 8, 3, 7, 4, 6, 10, 2]

It can easily also yield

Sequence B: [1, 10, 6, 4, 7, 3, 8, 5, 9, 2]

To determine range per variable, we first get its global max and min determined by its initial condition. Trivially, for 1 that is [0,1], and for 2 that is [0.5, 1], for 3 [0.33, 0.67]. Every condition after that reduces the global max and min of prior variables, yielding its local max and min. For example given the sequence [1,3,4,2], with 4 being [0.5, 0.75], 2 becomes [0.75,1], 3 becomes [0.33, 0.5], and 1 becomes [0,0.25]. Notice that there is a gap in [0.25, 0.33]. This is due to the global min restriction of 3rd variable - even though it can occupy [0.25, 0.5] by virtue of its order in the sequence, its acceptable range can only be [0.33, 0.5]. 


In [32]:
def order_seq(n):
    assert n>0, 'must input at least one interval'
    if n == 1:
        return [n]
    lst = [1,2]
    c = 3
    while c <= n:
        if c%2==0:
            i = lst.index((c+2)/2)
            lst.insert(i+1, c)
        else:
            i = lst.index((c+1)/2)
            lst.insert(i, c)
        c+=1
    return lst

In [33]:
def order_seq_plus(n:int, start:int, end:int):
    assert n>0, 'must input at least one interval'
    # total length
    total = start+end
    # dict to hold var ranges
    dct = {
        1: [0,0.5],
        2: [0.5,1]
    }
    # list for ordering
    lst = [1,2]
    # start at 3rd condition
    if n == 1:
        return [n]
    c = 3
    while c <= n:
        if c%2==0:
            i = lst.index((c+2)/2)
            lst.insert(i+1, c)
        else:
            i = lst.index((c+1)/2)
            lst.insert(i, c)
        interval = total/c
        g_min = lst.index(c)*interval
        g_max = g_min+interval
        dct[c] = [g_min, g_max]
        c+=1
    return lst, dct

In [34]:
order_seq(10)

[1, 9, 5, 8, 3, 7, 4, 6, 10, 2]

In [35]:
order_seq_plus(10, 0, 1)

([1, 9, 5, 8, 3, 7, 4, 6, 10, 2],
 {1: [0, 0.5],
  2: [0.5, 1],
  3: [0.3333333333333333, 0.6666666666666666],
  4: [0.5, 0.75],
  5: [0.2, 0.4],
  6: [0.6666666666666666, 0.8333333333333333],
  7: [0.42857142857142855, 0.5714285714285714],
  8: [0.25, 0.375],
  9: [0.1111111111111111, 0.2222222222222222],
  10: [0.8, 0.9]})